### ISSUE TITLE
[wip] Update Falcor Usage

### ISSUE THREAD

**Issue by joshdmiller** [StoryShop/app#55]
We currently have some unnecessary re-renders and too many listeners to the store.

We should migrate from managing falcor through a behaviour to using a HOC. We should further remove the independent re-rendering that the behaviour does in favour of a global re-render through a redux-like `FalcorProvider`.

@sethmerrick and @samcorcos: Thoughts?

**Comment by sethmerrick** [comment]
@joshdmiller I agree the effort has value in improving existing performance and setting a cleaner stage as we prepare to add several more dynamic components.
